vkcs/mlplatform/resource_jupyterhub.go

```diff
@@ -0,0 +1,448 @@
+package mlplatform
+
+import (
+	"context"
+	"fmt"
+	"sort"
+
+	"golang.org/x/exp/maps"
+
+	"github.com/hashicorp/terraform-plugin-framework-timeouts/resource/timeouts"
+	"github.com/hashicorp/terraform-plugin-framework/path"
+	"github.com/hashicorp/terraform-plugin-framework/resource"
+	"github.com/hashicorp/terraform-plugin-framework/resource/schema"
+	"github.com/hashicorp/terraform-plugin-framework/resource/schema/planmodifier"
+	"github.com/hashicorp/terraform-plugin-framework/resource/schema/stringplanmodifier"
+	"github.com/hashicorp/terraform-plugin-framework/types"
+	"github.com/hashicorp/terraform-plugin-log/tflog"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/retry"
+	"github.com/vk-cs/terraform-provider-vkcs/vkcs/internal/clients"
+	"github.com/vk-cs/terraform-provider-vkcs/vkcs/internal/services/mlplatform/v1/dnsnames"
+	"github.com/vk-cs/terraform-provider-vkcs/vkcs/internal/services/mlplatform/v1/instances"
+	"github.com/vk-cs/terraform-provider-vkcs/vkcs/internal/util"
+	"github.com/vk-cs/terraform-provider-vkcs/vkcs/internal/util/errutil"
+)
+
+// Ensure the implementation satisfies the desired interfaces.
+var _ resource.Resource = &JupyterHubResource{}
+
+func NewJupyterHubResource() resource.Resource {
+	return &JupyterHubResource{}
+}
+
+type JupyterHubResource struct {
+	config clients.Config
+}
+
+func (r *JupyterHubResource) Metadata(ctx context.Context, req resource.MetadataRequest, resp *resource.MetadataResponse) {
+	resp.TypeName = "vkcs_mlplatform_jupyterhub"
+}
+
+type JupyterHubResourceModel struct {
+	ID               types.String              `tfsdk:"id"`
+	Name             types.String              `tfsdk:"name"`
+	DomainName       types.String              `tfsdk:"domain_name"`
+	AdminName        types.String              `tfsdk:"admin_name"`
+	AdminPassword    types.String              `tfsdk:"admin_password"`
+	FlavorID         types.String              `tfsdk:"flavor_id"`
+	AvailabilityZone types.String              `tfsdk:"availability_zone"`
+	BootVolume       *MLPlatformVolumeModel    `tfsdk:"boot_volume"`
+	DataVolumes      []*MLPlatformVolumeModel  `tfsdk:"data_volumes"`
+	Networks         []*MLPlatformNetworkModel `tfsdk:"networks"`
+	S3FSBucket       types.String              `tfsdk:"s3fs_bucket"`
+	CreatedAt        types.String              `tfsdk:"created_at"`
+	PrivateIP        types.String              `tfsdk:"private_ip"`
+	DNSName          types.String              `tfsdk:"dns_name"`
+	Region           types.String              `tfsdk:"region"`
+	Timeouts         timeouts.Value            `tfsdk:"timeouts"`
+}
+
+func getSchemaJupyterHub(ctx context.Context, resp *resource.SchemaResponse) map[string]schema.Attribute {
+	jupyterHubAttrs := map[string]schema.Attribute{
+		"domain_name": schema.StringAttribute{
+			Optional:    true,
```

Comment:
do not we need here Compute = true as well?

Reply:
This field is used only for creation; result is stored in dns_name attribute